homework-selects/sql_questions.sql

```diff
@@ -4,54 +4,284 @@ SELECT first_name, last_name, salary, job_title
 FROM employees
          LEFT JOIN jobs USING (job_id);
 -- 2. the first and last name, department, city, and state province for each employee.
+SELECT FIRST_NAME, LAST_NAME, DEPARTMENT_NAME, STATE_PROVINCE
+FROM DEPARTMENTS
+         LEFT JOIN EMPLOYEES E on DEPARTMENTS.DEPARTMENT_ID = E.DEPARTMENT_ID
+         LEFT JOIN LOCATIONS L on DEPARTMENTS.LOCATION_ID = L.LOCATION_ID;
 -- 3. the first name, last name, department number and department name, for all employees for departments 80 or 40.
+SELECT FIRST_NAME, LAST_NAME, DEPARTMENT_ID, DEPARTMENT_NAME
+FROM EMPLOYEES
+         LEFT JOIN DEPARTMENTS
+                   USING (DEPARTMENT_ID)
+WHERE DEPARTMENT_ID = 40
+   or DEPARTMENT_ID = 80;
 -- 4. those employees who contain a letter z to their first name and also display their last name, department, city, and state province.
+SELECT first_name, last_name, department_name, city, state_province
+FROM employees e
+         LEFT JOIN departments d on d.department_id = e.department_id
+         LEFT JOIN locations l on d.location_id = l.location_id
+WHERE first_name like '%z%';
 -- 5. the first and last name and salary for those employees who earn less than the employee earn whose number is 182.
+SELECT first_name, last_name, salary
+FROM employees e
+WHERE e.salary < (SELECT salary
+                  FROM employees
+                  where employee_id = 182);
 -- 6. the first name of all employees including the first name of their manager.
+SELECT E.FIRST_NAME, M.FIRST_NAME AS MANAGER_NAME
+FROM EMPLOYEES E
+         LEFT JOIN EMPLOYEES M ON E.MANAGER_ID = M.EMPLOYEE_ID
+WHERE E.MANAGER_ID IS NOT NULL;
```

Comment:
for what case do we need 'WHERE E.MANAGER_ID IS NOT NULL' ?

Reply:
exist a person who don't have any manager

Reply:
perfect case to use INNER JOIN ))

Reply:
fixed